include/deal.II/matrix_free/mapping_info.h

```diff
@@ -454,6 +454,13 @@ namespace internal
        */
       SmartPointer<const Mapping<dim>> mapping;
 
+      /**
+       * Face QCollections related to each quadrature and active quadrature
+       * index.
+       */
+      std::vector<std::vector<dealii::hp::QCollection<dim - 1>>>
+        face_q_collection;
```

Comment:
I do not think this is the right place for this. We have tried to collect all data specific to the hp case into the `descriptor` class. Now this is a slightly different use to the wedge/pyramid case, but then it should be in `MappingInfoStorage` and not in the main class.

Reply:
I don't say that it is perfect as it is now but I don't think `MappingInfoStorage` is the right place. The information is directly liked to `MappingInfo::cell_data` and not to `MappingInfo::face_data`/`MappingInfo::face_data_by_cells`. 

Maybe PR #11401, where I have completely refactored the setup routine (and am hopefully close to getting face integrals running for wedges/pyramids), can demonstrate the relations more clearly:

https://github.com/dealii/dealii/blob/e88f1debe12f61e091b3893db6eb88025fd730c9/include/deal.II/matrix_free/mapping_info.templates.h#L365-L445

Reply:
Alternatively, we could move `face_q_collection` to `MappingInfoStorage` and keep it in `MappingInfo::face_data` empty, or don't save and pass the information directly to relevant functions (in the assumption that it will never be used!?).

Reply:
Please not also the need for the reference-cell type, which can be later passed to a `FE_Nothing` object (see also PR #11403).

Reply:
Yes, I think putting it in `MappingInfoStorage` but leaving things empty for the cases where it makes no sense (say `face_data`).

Reply:
OK. I have changed it.